src/webparts/laserficheAdminConfiguration/components/ManageMappingsPage/ManageMappingsPage.tsx

```diff
@@ -98,94 +115,65 @@ export default function ManageMappingsPage(props: IManageMappingsPageProps) {
       setValidationMessage(laserficheValidationMapping);
     } else {
-      GetItemIdByTitle().then((results: IListItem[]) => {
-        if (results != null) {
-          let entryExists = false;
-          const itemId = results[0].Id;
-          const jsonValue = JSON.parse(results[0].JsonValue);
-          if (jsonValue.length > 0) {
-            for (let i = 0; i < jsonValue.length; i++) {
-              if (jsonValue[i].id == rows[idx].id) {
-                entryExists = true;
-                break;
-              }
-            }
-            if (entryExists == true) {
-              UpdateExistingMapping(jsonValue, rows, idx, itemId);
-            } else {
-              AddNewInExistingMapping(jsonValue, rows, idx, itemId);
-            }
+      const existingMappings: {
+        id: string;
+        mappings: ProfileMappingConfiguration[];
+      } = await getManageMappingsAsync();
+      if (existingMappings != null) {
+        if (existingMappings.mappings.length > 0) {
+          const mappingExists = existingMappings.mappings.find(
+            (mapping) => mapping.id === rows[idx].id
+          );
+          if (mappingExists !== undefined) {
+            await updateExistingMappingAsync(
+              existingMappings.mappings,
+              rows,
+              idx,
+              existingMappings.id
+            );
           } else {
-            const restApiUrl = `${getSPListURL(
-              props.context,
-              ADMIN_CONFIGURATION_LIST
-            )}/items(${itemId})`;
-            const row = [...mappingRows];
-            const newjsonValue = [
-              {
-                id: row[idx].id,
-                SharePointContentType: row[idx].SharePointContentType,
-                LaserficheContentType: row[idx].LaserficheContentType,
-                toggle: true,
-              },
-            ];
-            const jsonObject = JSON.stringify(newjsonValue);
-            const body: string = JSON.stringify({
-              Title: MANAGE_MAPPING,
-              JsonValue: jsonObject,
-            });
-            const options: ISPHttpClientOptions = {
-              headers: {
-                Accept: 'application/json;odata=nometadata',
-                'content-type': 'application/json;odata=nometadata',
-                'odata-version': '',
-                'IF-MATCH': '*',
-                'X-HTTP-Method': 'MERGE',
-              },
-              body: body,
-            };
-            props.context.spHttpClient.post(
-              restApiUrl,
-              SPHttpClient.configurations.v1,
-              options
+            await addNewInExistingMappingAsync(
+              existingMappings.mappings,
+              rows,
+              idx,
+              existingMappings.id
             );
-            rows[idx].toggle = !rows[idx].toggle;
-            setMappingRows(rows);
           }
         } else {
           const restApiUrl = `${getSPListURL(
             props.context,
```

Comment:
it will be more readable to refactor the two else case block into helper functions

Reply:
I'm going to ignore the comments on the ManageMappingsPage for now because I am doing a complete refactor in the next PR (id and toggle won't exist)